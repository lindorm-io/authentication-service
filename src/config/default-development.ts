import { IConfiguration } from "../typing";
import { NodeEnvironment } from "@lindorm-io/core";

export const configuration: IConfiguration = {
  NODE_ENVIRONMENT: NodeEnvironment.DEVELOPMENT,
  SERVER_PORT: 3000,
  HOST: "http://localhost/",

  JWT_ISSUER: "https://dev.lindorm.io/",
  JWT_ACCESS_TOKEN_EXPIRY: "1 minutes",
  JWT_AUTHORIZATION_TOKEN_EXPIRY: "30 minutes",
  JWT_IDENTITY_TOKEN_EXPIRY: "5 minutes",
  JWT_MULTI_FACTOR_TOKEN_EXPIRY: "30 minutes",
  JWT_REFRESH_TOKEN_EXPIRY: "5 minutes",

  CRYPTO_AES_SECRET: "secret",
  CRYPTO_SHA_SECRET: "secret",

  ACCOUNT_OTP_ISSUER: "dev.lindorm.io",

  MAILGUN_API_KEY: "api-key",
  MAILGUN_DOMAIN: "dev.lindorm.io",
  MAILGUN_FROM: "noreply@lindorm.io",

  MONGO_INITDB_ROOT_USERNAME: "root",
  MONGO_INITDB_ROOT_PASSWORD: "password",
  MONGO_HOST: "localhost",
  MONGO_EXPOSE_PORT: 27017,
  MONGO_DB_NAME: "authentication",
};
