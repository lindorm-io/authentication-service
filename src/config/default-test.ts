import { IConfiguration } from "../typing";
import { NodeEnvironment } from "@lindorm-io/core";

export const configuration: IConfiguration = {
  NODE_ENVIRONMENT: NodeEnvironment.DEVELOPMENT,
  SERVER_PORT: 3000,
  HOST: "http://localhost/",

  JWT_ISSUER: "https://test.lindorm.io/",
  JWT_ACCESS_TOKEN_EXPIRY: "2 minutes",
  JWT_AUTHORIZATION_TOKEN_EXPIRY: "45 minutes",
  JWT_IDENTITY_TOKEN_EXPIRY: "1 days",
  JWT_MULTI_FACTOR_TOKEN_EXPIRY: "30 minutes",
  JWT_REFRESH_TOKEN_EXPIRY: "1 days",

  CRYPTO_AES_SECRET: "secret",
  CRYPTO_SHA_SECRET: "secret",

  ACCOUNT_OTP_ISSUER: "test.lindorm.io",

  MAILGUN_API_KEY: "api-key",
  MAILGUN_DOMAIN: "test.lindorm.io",
  MAILGUN_FROM: "noreply@lindorm.io",

  MONGO_INITDB_ROOT_USERNAME: "root",
  MONGO_INITDB_ROOT_PASSWORD: "password",
  MONGO_HOST: "localhost",
  MONGO_EXPOSE_PORT: 27017,
  MONGO_DB_NAME: "authentication",
};
