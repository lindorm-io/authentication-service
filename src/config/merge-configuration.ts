import { IConfiguration } from "../typing";
import { TObject } from "@lindorm-io/core";

export const mergeConfiguration = (primary: IConfiguration, secondary: IConfiguration): IConfiguration => {
  const object: TObject<any> = secondary;

  for (const [key, value] of Object.entries(primary)) {
    if (value === null || value === undefined) continue;
    object[key] = value;
  }

  const result: unknown = object;
  return result as IConfiguration;
};
