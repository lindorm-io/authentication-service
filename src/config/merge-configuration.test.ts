import { mergeConfiguration } from "./merge-configuration";
import { configuration as defaultDevelopment } from "./default-development";
import { configuration as defaultTest } from "./default-test";

describe("mergeConfig", () => {
  test("should prioritize primary input", () => {
    expect(mergeConfiguration(defaultTest, defaultDevelopment)).toMatchSnapshot();
  });
});
