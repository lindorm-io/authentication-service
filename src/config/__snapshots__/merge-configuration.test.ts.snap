// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mergeConfig should prioritize primary input 1`] = `
Object {
  "ACCOUNT_OTP_ISSUER": "test.lindorm.io",
  "CRYPTO_AES_SECRET": "secret",
  "CRYPTO_SHA_SECRET": "secret",
  "HOST": "http://localhost/",
  "JWT_ACCESS_TOKEN_EXPIRY": "2 minutes",
  "JWT_AUTHORIZATION_TOKEN_EXPIRY": "45 minutes",
  "JWT_IDENTITY_TOKEN_EXPIRY": "1 days",
  "JWT_ISSUER": "https://test.lindorm.io/",
  "JWT_MULTI_FACTOR_TOKEN_EXPIRY": "30 minutes",
  "JWT_REFRESH_TOKEN_EXPIRY": "1 days",
  "MAILGUN_API_KEY": "api-key",
  "MAILGUN_DOMAIN": "test.lindorm.io",
  "MAILGUN_FROM": "noreply@lindorm.io",
  "MONGO_DB_NAME": "authentication",
  "MONGO_EXPOSE_PORT": 27017,
  "MONGO_HOST": "localhost",
  "MONGO_INITDB_ROOT_PASSWORD": "password",
  "MONGO_INITDB_ROOT_USERNAME": "root",
  "NODE_ENVIRONMENT": "test",
  "REDIS_PORT": 6379,
  "SERVER_PORT": 3000,
}
`;
